shared/src/main/scala-3/zipper/contrib/shapeless3/Replacer.scala

```diff
@@ -0,0 +1,28 @@
+package contrib.shapeless3
+
+trait Replacer[L <: Tuple, U, V]:
```

Comment:
As a matter of fact, the full-fledged migration of shapeless-2 to Scala 3 is currently in progress (https://github.com/milessabin/shapeless/pull/1200 — it's such a monumental work, I'm just shocked). So we need to migrate `Replacer` and `Selector` ourselves for a while.

Reply:
But yeah, I think some clarifications should be put into the comments. I'll go over it tomorrow.

Reply:
I wonder if the native derivation capabilities of Scala 3 are sufficiently powerful to eschew shapeless altogether?

Reply:
I can imagine a solution powered by a Macro. However, it should also work with Scala 2 Macros. But I don't know if Scala 3 has anything out of pocket for replacing `shapeless.Generic`.

Reply:
I meant this: https://docs.scala-lang.org/scala3/reference/contextual/derivation.html#how-to-write-a-type-class-derived-method-using-low-level-mechanisms-1. But after a closer look it does not seem powerful enough (yet?). And I don’t like that using it seems to require writing a lot of `asInstanceOf` code.